lib-multisrc/libgroup/src/eu/kanade/tachiyomi/multisrc/libgroup/LibGroup.kt

```diff
@@ -311,8 +311,10 @@ abstract class LibGroup(
         }
 
         val sortingList = preferences.getString(SORTING_PREF, "ms_mixing")
-        val defaultBranchId = runCatching { getDefaultBranch(slugUrl.substringBefore("-")).first().id }.getOrNull()
-
+        var defaultBranchId: Int? = null
+        if (chaptersData.data.filter { it.branchesCount > 1 }.isNotEmpty()) { // excess request if branchesCount is only alone = slow update library witch rateLimitHost(apiDomain.toHttpUrl(), 1)
```

Comment:
This may break ? LibGroup sites itself always making calls to get default branch even if chapters have only one branch.

It would be better to get branch list length, than trust `branchesCount`. Also getting branches count should be implemented in `Chapter` class in `LibGroupDto.kt` file.

No need `defaultBranchId` to be `var`, make it `val` and move if to definition
`val defaultBranchId: Int? = if(...) {...} else null`

Reply:
> This may break in future ? LibGroup sites itself always making calls to get default branch even if chapters have only one branch.

In the future, we'll see, in this case it's just an extra request for emptiness and on the site, in fact, too

Reply:
and mangas with several branches is few